doc/howtos/rdtraining/B_acl_irrules.rst

```diff
@@ -4,4 +4,424 @@
+
+.. danger::
+
+    These features should generally be avoided, and only used with extreme care,
+    after having checked that the current user and operation should be able to
+    bypass normal access rights validation.
+
+    Operations performed in such modes should also rely on user input as little
+    as possible, and should validate it to the maximum extent they can.
+
+.. exercise::
+
+    Having to give agents "billing" access so they can confirm a property sale
+    when ``estate_account`` is installed is giving them unnecessary access.
+
+    Update ``estate_account`` to bypass access rights and rules when creating
+    the invoice, but only if they are allowed to modify the property.
+
+.. _howto/rdtraining/B_acl_irrules/multicompany:
+
+Multi-company security
+======================
+
+Odoo can be used to manage multiple companies in the same database, this can
+be useful to unify workflows, facilitate exchanges, because the companies are
+separate for legal or regulatory reasons but conceptually the same entity, ...
+
+As such it usually makes sense to figure out how the module fits in a
+multi-company setup: some would ignore it (all companies have access to all
+data) while others would completely silo data and operations (no cross-company
+operations or data sharing), or every scenario inbetween and to the side.
+
+As originally noted, it often makes sense to default to the more restrictive
+set-up, as it is easier for users to disable the relevant rules than it is to
+create them.
+
+This means identifying company-specific data, and isolating companies from one
+another, mostly through *multi-company access rules*.
+
+Which just means access rules setting based on the ``company_id`` or
+``company_ids`` variables described above: in Odoo, a user is associated with
+any number of companies to which they have access, but they can only "work in"
+a single company at any point, this is what those two variable means,
+``company_id`` is the company currently selected by the user while
+``company_ids`` is all the companies the user has access to.
+
+The most common rule is to simply check if a company configured on a record
+matches one of the user's companies:
+
+.. code-block:: xml
+
+    <record model="ir.rule" id="hr_appraisal_plan_comp_rule">
+        <field name="name">Appraisal Plan multi-company</field>
+        <field name="model_id" ref="model_hr_appraisal_plan"/>
+        <field name="domain_force">[
+            '|', ('company_id', '=', False),
+                 ('company_id', 'in', company_ids)
+        ]</field>
+    </record>
+
+If the record *must* have a company configured for one reason or an other, then
+the first criterion is unnecessary.
+
+.. danger::
+
+    Multi-company rules are usually :ref:`global <howto/rdtraining/B_acl_irrules/rules/global>`,
+    otherwise there is a high risk that additional rules would allow bypassing
+    the muti-company rules (though that is sometimes desired).
+
```

Comment:
I'd add several steps in the exercise:
1. Create a new company, with an estate agent in each company. The manager should be in both companies.
2. Create a few properties with each estate agent, in each company
3. Whatever is the company of the manager, he can see all the properties => That's not good for our use case
4. Add a `company_id` field on the `estate.property` model
5. Add the record rules

I think all these steps fit in your explanation. Steps 1-3 could go just before the "Which just means access rules setting based on the..." part